### PR TITLE
ThrusterServer::start -> ThrusterServer::build

### PR DESCRIPTION
**Pull request by vorot93**

### CODE REVIEW

File: thruster-app/src/app.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ fn _add_method_to_route_from_str(method: &str, path: &str) -> String {
 /// There are three main parts to creating a thruster app:
 /// 1. Use `App.create` to create a new app with a custom context generator
 /// 2. Add routes and middleware via `.get`, `.post`, etc.
-/// 3. Start the app with `App.start`
+/// 3. Build the app future with `App.build` and spawn it on the executor
```
Comment (trezm) [inline thruster-app/src/app.rs]: Let's keep the original documentation with the addition of _"or build the app future with `App.build` and spawn it on an executor."_

File: thruster-server/src/server.rs [hunk 2]
```diff
@@ -14,93 +14,13 @@ use thruster_core::http::Http;
...
-// #[cfg(not(windows))]
-// use net2::unix::UnixTcpBuilderExt;
-
 use crate::thruster_server::ThrusterServer;
 
 pub struct Server<T: 'static + Context<Response = Response> + Send> {
     app: App<Request, T>,
 }
 
-// impl<T: 'static + Context<Response = Response> + Send> Server<T> {
```
Comment (trezm) [inline thruster-server/src/server.rs]: I have a todo to actually fix this code and uncomment it -- keep it around for the time being as (when it's running correctly,) has a large impact on benchmarks.

File: thruster/examples/hyper_most_basic.rs [hunk 3]
```diff
@@ -14,12 +14,13 @@ async fn plaintext(mut context: Ctx, _next: MiddlewareNext<Ctx>) -> Ctx {
     context
 }
 
-fn main() {
+#[tokio::main]
```
Comment (trezm) [inline thruster/examples/hyper_most_basic.rs]: Let's keep this and the hello world with the `start` syntax, as these should be as simple + easy as possible.

File: thruster/examples/hello_world.rs [hunk 4]
```diff
@@ -11,13 +11,14 @@ async fn plaintext(mut context: Ctx, _next: MiddlewareNext<Ctx>) -> Ctx {
     context
 }
 
-fn main() {
+#[tokio::main]
```
Comment (trezm) [inline thruster/examples/hello_world.rs]: Let's keep this and the most basic hyper with the `start` syntax, as these should be as simple + easy as possible.

**Comment by trezm** [comment]
@vorot93 I think you might have missed my earlier comment about keeping `start` around and simply calling into build in the method body:

> Instead of swapping out start for build, let's have start call build instead. Having a dead simple way to start the server is still an important feature.

Could you update that as well?

**Comment by vorot93** [comment]
Actually, it’s what it does now as a predefined trait method.

**Comment by trezm** [comment]
Didn't notice that -- that's awesome!! LGTM 👍

**Pull request merged**